modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/admin/throttlingpolicy/GetThrottlingPoliciesTestCase.java

```diff
@@ -0,0 +1,154 @@
+    private final String getThrottlePoliciesResource = "/throttling/policies/search";
+    private final  String appPolicyName="50PerMin";
+    private final String subPolicyName ="Gold";
+    private final String apiPolicyName="10KPerMin";
+    private String getThrottlePoliciesUrl;
+
+    @Factory(dataProvider = "userModeDataProvider")
+    public GetThrottlingPoliciesTestCase(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @DataProvider public static Object[][] userModeDataProvider() {
+        return new Object[][] { new Object[] { TestUserMode.SUPER_TENANT_ADMIN },
+                new Object[] { TestUserMode.TENANT_ADMIN } };
+    }
+
+    @BeforeClass(alwaysRun = true) public void setEnvironment() throws Exception {
+        super.init(userMode);
+        getThrottlePoliciesUrl =
+                adminURLHttps + APIMIntegrationConstants.REST_API_ADMIN_CONTEXT_FULL_0 + getThrottlePoliciesResource;
+    }
+
+    @Test(groups = { "wso2.am" }, description = "Get ThrottlePolicies List and check few default policies")
+    public void testThrottlePoliciesGet() throws Exception {
+        //construct get Throttle Policies url
+        URL getRequest = new URL(getThrottlePoliciesUrl);
+        ThrottlePolicyDetailsListDTO policies = getPolicies(user.getUserName(), user.getPassword(), getRequest);
+        List<ThrottlePolicyDetailsDTO> policyList = policies.getList();
+        Assert.assertTrue(containsPolicy(policyList, appPolicyName), "Doesn't contain app Policy");
+        Assert.assertTrue(containsPolicy(policyList, subPolicyName), "Doesn't contain sub Policy");
+        Assert.assertTrue(containsPolicy(policyList, apiPolicyName), "Doesn't contain api Policy");
+    }
+
+    /**
+     * Checks whether a certain policy exists in the policy list
+     * @param list throttle policy details list
+     * @param name throttle policy name
+     * @return true or false on the policy presence in the policy list
+     */
+    private boolean containsPolicy(List<ThrottlePolicyDetailsDTO> list, String name) {
+        return list.stream().anyMatch(o -> o.getPolicyName().equals(name));
+    }
+
+    /**
+     * Execute GET request to read the data to ThrottlePolicyDetailsListDTo
+     * @param getRequest url for get throttling policies
+     * @return ThrottlePolicyDetailsListDTo
+     * @throws URISyntaxException throws if URI is malformed
+     * @throws IOException throws if connection issues occurred
+     */
+    private ThrottlePolicyDetailsListDTO getPolicies(String username, String password, URL getRequest)
+            throws URISyntaxException, IOException {
+        //construct export Throttle Policy url
+        File TempDir = Files.createTempDir();
+        String fileName = "get_policies";
+        File policiesFile = new File(TempDir.getAbsolutePath() + File.separator + fileName + ".json");
+        CloseableHttpResponse response = getThrottlePoliciesRequest(getRequest, username, password);
```

Comment:
Can use generated api files for this instead

Reply:
Fixed with 3106d801961d2d8e82a5a35117427b8337a92455